bin/sync_vsc_users.py

```diff
@@ -85,97 +78,93 @@ def main():
         'start_timestamp': ('Timestamp to start the sync from', str, 'store', None),
     }
 
-    opts = ExtendedSimpleOption(options)
-    stats = {}
+    def do(self, dry_run):
+        """
+        Actual work
+        - build the filter
+        - fetches the users
+        - process the users
+        - write the new timestamp if everything went OK
+        - write the nagios check file
+        """
 
-    (last_timestamp, start_time) = retrieve_timestamp_with_default(
-        SYNC_TIMESTAMP_FILENAME,
-        start_timestamp=opts.options.start_timestamp)
-    logging.info("Using timestamp %s", last_timestamp)
-    logging.info("Using startime %s", start_time)
+        stats = {}
 
-    try:
-        client = AccountpageClient(token=opts.options.access_token, url=opts.options.account_page_url + "/api/")
+        #(last_timestamp, start_time) = retrieve_timestamp_with_default(
+            #SYNC_TIMESTAMP_FILENAME,
+            #start_timestamp=self.options.start_timestamp)
 
-        institute = opts.options.host_institute
+        institute = self.options.host_institute
 
         (users_ok, users_fail) = ([], [])
         (quota_ok, quota_fail) = ([], [])
-        if opts.options.user:
-            changed_accounts = client.account.institute[institute].modified[last_timestamp].get()[1]
+        if self.options.user:
+            changed_accounts, _ = self.apc.get_accounts()  # we ignore inactive accounts
```

Comment:
no institute nor timestamp passed?

Reply:
No, this mistaken care of by the Sync class.

Reply:
should be self.get_accounts, the timestamp is processed in sync class, not APclient

Reply:
and found no clue where the site/institute should have been taken care of